ledger-core-ethereum/src/ethereum/factories/EthereumLikeWalletFactory.cpp

```diff
@@ -36,36 +36,37 @@
 #include <core/api/ConfigurationDefaults.hpp>
 
 #include <ethereum/EthereumLikeWallet.hpp>
+#include <ethereum/EthereumNetworks.hpp>
+#include <ethereum/database/Migrations.hpp>
 #include <ethereum/explorers/LedgerApiEthereumLikeBlockchainExplorer.hpp>
 #include <ethereum/factories/EthereumLikeWalletFactory.hpp>
 #include <ethereum/factories/EthereumLikeKeychainFactory.hpp>
 #include <ethereum/observers/LedgerApiEthereumLikeBlockchainObserver.hpp>
 #include <ethereum/synchronizers/EthereumLikeBlockchainExplorerAccountSynchronizer.hpp>
-#include <ethereum/EthereumNetworks.hpp>
 
 #define STRING(key, def) entry.configuration->getString(key).value_or(def)
 
 namespace ledger {
     namespace core {
-        EthereumLikeWalletFactory::EthereumLikeWalletFactory(const api::Currency &currency,
-                                                             const std::shared_ptr<Services> &services):
-                                                            AbstractWalletFactory(currency, services) {
-                _keychainFactories = {{api::KeychainEngines::BIP49_P2SH, std::make_shared<EthereumLikeKeychainFactory>()}};
+        EthereumLikeWalletFactory::EthereumLikeWalletFactory(
+            const api::Currency &currency,
+            const std::shared_ptr<Services> &services
+        ):
+          AbstractWalletFactory(currency, services) {
+            _keychainFactories = {{api::KeychainEngines::BIP49_P2SH, std::make_shared<EthereumLikeKeychainFactory>()}};
+            // create the DB structure if not already created
+            services->getDatabaseSessionPool()->forwardMigration<EthereumMigration>();
+
+            // TODO: add currencies
```

Comment:
This will be done later and should be done for all coins (not needed right now and a Jira ticket is already present for that).